computation/src/main/java/com/powsybl/computation/ComputationExceptionBuilder.java

```diff
@@ -133,4 +157,23 @@ private ComputationExceptionBuilder readFileToMap(@Nullable Path path, Map<Strin
         }
         return this;
     }
+
+    /**
+     * Creates the computation exception.
+     *
+     * @return the created computation exception.
+     */
+    public ComputationException build() {
+        ComputationException exception;
+        if (message != null && cause != null) {
+            exception = new ComputationException(message, cause, outMsgByLogFileName, errMsgByLogFileName, bytesByFileName);
+        } else if (message != null) {
+            exception = new ComputationException(message, outMsgByLogFileName, errMsgByLogFileName, bytesByFileName);
+        } else if (cause != null) {
+            exception = new ComputationException(cause, outMsgByLogFileName, errMsgByLogFileName, bytesByFileName);
+        } else {
+            exception = new ComputationException(outMsgByLogFileName, errMsgByLogFileName, bytesByFileName);
+        }
+        return exception;
```

Comment:
To be discussed: is it necessary to handle all the possibilities? What if `ComputationException` accept Nullable message or cause?
You could simply write `new ComputationException(message, cause, outMsgByLogFileName, ...);`

Reply:
The `cause` could be nullable, but it's not that easy for the `message` if we want to keep the usual behaviour :
`new Throwable(null, cause)` and `new Throwable(cause)` are not equivalent, the second one takes the description of the cause as its message while the first one really does not have any message.
And we cannot move the else/if switches to the constructor and choose here which overload of the `super` constructor is called, since it must be the first line of the constructor.

So in order to keep the usual construction possibilities offered by exceptions in the JDK, I think it's better to either:
* keep it this way
* make only the `cause` nullable
* make both nullable in the constructor, but handle the case null message + non null cause in the builder, to fill the message with the cause, as is usually done by `Throwable(cause)`
What possibility would you prefer?

Reply:
I would keep it as it is. It make sense to me to stick to existing Exception constructors.